### DISCUSSION TITLE
The code that you entered has expired.

### DISCUSSION THREAD

**Discussion by thedavegray** [lavinir/hassio-onedrive-backup#46] [General]
Hi

Having issues logging into OneDrive using the code given by the page: http://********:8123/hassio/addon/de91e161_hassio_onedrive_backup/logs

I have tried three times but get the same error.  Steps are: 

- Copy the activation code 
- Open https://microsoft.com/devicelogin and enter the code 
- Enter the email address associated with OneDrive
- This sends a request to my MS Authenticator (choose one of three numbers)
- Prompted for app access (hassion-onedrive-backup)
- Click Yes

![image](https://user-images.githubusercontent.com/4395873/217790138-47faded5-9289-4d51-a6f4-10173710f1e6.png)

The first error in the log says this 
`02/09/2023 10:00:10 [11] Error: Unexpected error. Azure.Identity.AuthenticationFailedException: DeviceCodeCredential authentication failed: Verification code expired before contacting the server`

It seems like the activation code taken from ...backup/logs has expired.  I can't see how to create a new code except 
rebooting the Raspberry Pi that HA runs on, which I have done three times.  Same result each time. 

Hoping someone has seen this before

Thanks

**Comment by lavinir** [comment]
You are getting the same code each time ? 
Try to uninstall and re-install the add-on and re-authenticate.

**Comment by thedavegray** [comment]
Hi,
I get a different code if I reboot the Pi, none of these have worked
though.
I tried the Google drive backup which works fine, but will try the OneDrive
again at the weekend.

Thanks

**Reply by lavinir** [reply]
There are no similarities in the authentication flow with the google drive addon :)
The way the addon uses authentication is using Microsoft's official library and I'm not sure what's happening for you but I would try this:
1. Uninstall the Add-on
2. Re-install
3. Open authentication link in incognito mode in the browser and attempt authnetication.

**Comment by thedavegray** [comment]
Hi
Thanks for the info.

I think I will just use the Google drive for backups as this was easy to
do.

Regards

**Comment by housemaister** [comment]
I had the same issue. 
After restarting the add-on, I got a new code, which then worked.

**Reply by isvec** [reply]
Thanks @housemaister, that was precisely what helped me. A simple restart of the addon.
